integration/spark/src/main/java/openlineage/spark/agent/lifecycle/plan/visitor/DatasetSourceVisitor.java

```diff
@@ -61,15 +62,39 @@ public boolean isDefinedAt(LogicalPlan x) {
   }
 
   @Override
-  public List<LineageEvent.Dataset> apply(LogicalPlan x) {
+  public List<OpenLineage.Dataset> apply(LogicalPlan x) {
     DatasetSource datasetSource =
         findDatasetSource(x)
             .orElseThrow(() -> new RuntimeException("Couldn't find DatasetSource in plan " + x));
     return Collections.singletonList(
-        LineageEvent.Dataset.builder()
-            .name(datasetSource.name())
-            .namespace(datasetSource.namespace())
-            .facets(PlanUtils.datasetFacet(x.schema(), datasetSource.namespace()))
-            .build());
+        PlanUtils.getDataset(
+            datasetSource.name(),
+            datasetSource.namespace(),
+            PlanUtils.datasetFacet(x.schema(), datasetSource.namespace())));
+  }
+
+  /**
+   * DatasetSource is an interface that allows instrumenting an existing Relation or Datasource
+   * class with the ability to report its OpenLineage namespace and name. Relations that intend to
+   * participate in the OpenLineage reporting can implement this interface directly and they will be
+   * found in the Spark {@link LogicalPlan} during plan execution. Otherwise, {@link
+   * java.lang.instrument.ClassFileTransformer}s can be instrumented to dynamically rewrite classes
+   * to implement this interface with logic to return the necessary data. In that case, the
+   * instrumented class will still be found in the {@link LogicalPlan} during execution and the
+   * datasource details will be included in the lineage.
+   */
+  public interface DatasetSource {
```

Comment:
Why was this moved from the original file? For developers who might plan to implement this directly, I think it makes more sense to keep it as its own file.

Reply:
my bad, I was thinking that it's used in one place, but it's not,next time will search by name not only by usages